### PR TITLE
Use NVD_GPU to control which DRM node to open in direct backend mode

### PR DESCRIPTION
**Pull request by philipl**
When running in direct backend mode, we need to open the DRM node ourselves, and the first node may not be the right one.

This change uses NVD_GPU to identify which one to use. It might be desirable to use a different env var because it will count non-nvidia GPUs so it could be confusing. That's easy to change.

### CODE REVIEW

**Comment by elFarto** [comment]
Hmm, I swear I wrote a comment on this yesterday, Github seems to have eaten it.

I think I'd prefer a `NVD_DRM_DEVICE=/dev/dri/renderD129` rather than repurpose NVD_GPU, which is currently used to refer to which CUDA GPU to use.

With that said, I'd ideally like some way to detect the DRM device from the X11 handle that's passed in (we should have a DRM device if using Wayland), but the only ways I currently know of doing that are using the DRI3 Open method, which wasn't supported the last time I looked, or some wacky NV-GLX calls.

I guess we could just scan for the first NVIDIA DRM device, and leave more complicated setups to the NVD_DRM_DEVICE variable.

**Comment by philipl** [comment]
How about using NVD_GPU to mean the nth nvidia device, and we iterate over the render nodes and look for nvidia devices specifically. Then we can keep using the same variable with it meaning the same thing.

**Comment by philipl** [comment]
Updated diff to use `NVD_GPU` to specifically pick the nth nvidia gpu and ignore others.

File: src/direct/direct-export-buf.c [hunk 1]
```diff
@@ -54,9 +54,44 @@ bool direct_initExporter(NVDriver *drv) {
 
     //make sure we have a drm fd
     if (drv->drmFd == -1) {
-        //TODO make this configurable
-        drv->drmFd = open("/dev/dri/renderD128", O_RDWR|O_CLOEXEC);
-        LOG("Manually opened DRM device");
+        uint8_t nvdGpu = 0;
+        char *nvdGpuStr = getenv("NVD_GPU");
```
Comment (elFarto) [inline src/direct/direct-export-buf.c:58]: This value is already be available as `drv->cudaGpuId`. That variable might be set to `-1`, meaning the default GPU, in this case we can change it to `0`.

**Comment by philipl** [comment]
Updated to use `drv->cudaGpuId`

**Comment by elFarto** [comment]
Ok, that looks good. Thanks for that!

**Pull request merged**